dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java

```diff
@@ -82,7 +114,257 @@ public Bundle getBundleById(String id) throws DotDataException {
+
+				bundleIds.forEachOrdered(bundleId -> {
+
+					this.internalDeleteBundleAndDependenciesThrowRuntime(bundleId, user);
+					bundlesDeleted.add(bundleId);
+				});
+			}
+		} catch (IOException e) {
+
+			throw new DotDataException(e);
+		}
+
+		return bundlesDeleted.build();
+	} // deleteAllBundles.
+
+	private Stream<String> getAllBundleIds(final List<Status> statusList, final String userId, final boolean isAdmin)
+			throws IOException {
+
+		return this.getBundleIds(
+				(Integer limit, Integer offset) -> {
+
+					try {
+
+						return isAdmin?
+								this.publishAuditAPI.getBundleIdByStatus(statusList, limit, offset) :
+								this.publishAuditAPI.getBundleIdByStatusFilterByOwner(statusList, limit, offset, userId);
+					} catch (DotDataException e) {
+
+						throw new DotRuntimeException(e);
+					}
+				},
+				(String bundleId) -> bundleId
+		);
+	}
+
+	@WrapInTransaction
+	@Override
+	public Set<String>  deleteAllBundles(final User user) throws DotDataException {
+
+		final ImmutableSet.Builder<String> bundlesDeleted = new ImmutableSet.Builder<>();
+		final boolean isAdmin = this.userAPI.isCMSAdmin(user);
+
+		Logger.info(this, "Deleting all bundles by user: " + user.getUserId());
+
+		try {
+
+			Logger.info(this, "Deleting all bundles by user: " + user.getUserId());
+			try (final Stream<String> bundleIds = this.getAllBundleIds(user.getUserId(), isAdmin)) {
+
+				bundleIds.forEachOrdered(bundleId -> {
+
+					this.internalDeleteBundleAndDependenciesThrowRuntime(bundleId, user);
+					bundlesDeleted.add(bundleId);
+				});
+			}
+		} catch (IOException e) {
+
+			throw new DotDataException(e);
+		}
+
+		return bundlesDeleted.build();
+	}
+
+	private Stream<String> getAllBundleIds(final String userId, final boolean isAdmin)
+			throws IOException {
+
+		return this.getBundleIds(
+				(Integer limit, Integer offset) -> {
+
+					try {
+						return isAdmin?
+								this.bundleFactory.findSentBundles(limit, offset):
+								this.bundleFactory.findSentBundles(userId, limit, offset);
+					} catch (DotDataException e) {
+
+						throw new DotRuntimeException(e);
+					}
+				},
+				Bundle::getId
+		);
+	}
+
+	/**
+	 * Returns a streamable list of bundles id, candidates to delete.
+	 * @param bundleFinder {@link BiFunction} this function will receive the limit and current offset and returns the list of T bundles
+	 * @param bundleToIdentifierConverter {@link Function} this function will convert the T bundle to bundle id as String
+	 * @param <T>
+	 * @return Stream of String (streamable list of bundles)
+	 * @throws IOException
+	 */
+	private <T> Stream<String> getBundleIds(final BiFunction<Integer, Integer, List<T>> bundleFinder,  // receives limit and offset and returns the list of T
+											final Function<T, String> bundleToIdentifierConverter)    // convert T to bundle id
+			throws IOException {
+
+		final File tempFile       = FileUtil.createTemporalFile("bundle-ids");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)